### PR TITLE
Fix freeimage 3.17 for Mac

### PR DESCRIPTION
**Pull request by jakirkham**
Tested on Mac OS 10.9 and CentOS 6.3.

* Added MD5 checksums for downloads.
* Overrides where includes and libraries are installed to ensure they go to the right directory on Linux (also works on Mac).
* `sed` in-place requires an extension be specified for the original files. Will still work on Linux.
* `sed` gets hung up on what it considers binary files. Must convince it that it can work through them.
* ~~One header file is missing an include.~~ Appears to have been resolved.
* ~~A C++ source file does not appropriately cast types.~~ Appears to have been resolved.
* The OS X Makefile is completely outdated. So it needed to be modernized.
    * Ancient `gcc` specified `4.0`. ~~Just use `gcc` with any version.~~ Sometimes the `gcc`-built library has issues finding this symbol `__ZNSt8__detail15_List_node_base11_M_transferEPS0_S1_`, but `clang` does not. As the `gcc` frontend is being phased, out just use `clang`. 
    * Build steps for ancient platforms namely PPC and 32-bit Mac. Drop from the Makefile.

### CODE REVIEW

**Comment by jakirkham** [comment]
Upon further review, I was able to eliminate nearly all of the patches. After all, most of these patches were needed on 3.15.4 and were merely ported to 3.17.0. The only remaining patch is for "Makefile.osx".

**Comment by jakirkham** [comment]
@almarklein, see response below.

> Thanks @jakirkham. There's quite a few patches in there. Would it be an idea to contribute these to the FreeImage project instead?

That probably is a good idea. However, it appears most of them weren't necessary after the update to 3.17.0.

> I know FI uses sourceforge and SVN (yuk)...Also see this discussion about (not) moving FI to GH.

It sounds like from the discussion referenced ( http://sourceforge.net/p/freeimage/discussion/36111/thread/f2fae857/ ) that they are using CVS. Unfortunately, I don't know how to use this form of revision control and feel I don't have the bandwidth to pick it up, especially, as it appears to be dead. I have seen some forks of FreeImage on GitHub, but am not convinced any one of them is official and many appear to have deviated significantly.

> ...but that we would not have to verify that our patches work every time we update.

Currently, we have only one patch, which is for `Makefile.osx`. In this case, they are holding on to PPC and i386 support, which we stripped out. The original changes came out of a discussion on a FreeImage bug ( https://sourceforge.net/p/freeimage/bugs/228/ ). So, they are at least aware of this and have decided not to incorporate it and there is little we can do here.

<br>
In short, I generally agree with you that the changes should move upstream. However, due to the choices made by the FreeImage team, I don't think I am the person to ask. Regardless, the change offered here is a problem they are aware of and have chosen not to address. In short, I think there is little we can do in this case.

**Comment by almarklein** [comment]
Ok, I agree with you. Is this PR ready to go?

**Comment by jakirkham** [comment]
Yep, its ready. I have tested on Mac OS 10.9 and CentOS 6.3 with the most recent commit.
